x-pack/plugins/serverless_search/public/application/components/overview_panels/select_client.tsx

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiImage,
+  EuiLink,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import React from 'react';
+import { PLUGIN_ID } from '../../../../common';
+
+import { useKibanaServices } from '../../hooks/use_kibana';
+import { LanguageDefinition } from '../languages/types';
+import { OverviewPanel } from './overview_panel';
+import { docLinks } from '../../../../common/doc_links';
+interface SelectClientProps {
+  setSelectedLanguage: (language: LanguageDefinition) => void;
+  languages: LanguageDefinition[];
+  selectedLanguage: LanguageDefinition;
+}
+export const SelectClientPanel: React.FC<SelectClientProps> = ({
+  setSelectedLanguage,
+  languages,
+  selectedLanguage,
+}) => {
+  const { http } = useKibanaServices();
+  const { euiTheme } = useEuiTheme();
+
+  const panelItems = languages.map((language) => (
+    <EuiPanel
```

Comment:
It would be better to refactor this into a component instead of building this a list of items. There is a React performance gain to be had there too I believe.

Reply:
Implemented in  d6d7c092cff83cfaed661432541e8cccd85ab14c